src/Libraries/Microsoft.Extensions.DataIngestion/Processors/ClassificationEnricher.cs

```diff
@@ -0,0 +1,129 @@
+    {
+        _ = Throw.IfNull(chunks);
+
+        await foreach (IngestionChunk<string> chunk in chunks.WithCancellation(cancellationToken))
+        {
+            var response = await _chatClient.GetResponseAsync(
+            [
+                new(ChatRole.User, chunk.Content)
+            ], _chatOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            chunk.Metadata[MetadataKey] = _predefinedClasses.Contains(response.Text)
+                ? response.Text
+                : throw new InvalidOperationException($"Classification returned an unexpected class: '{response.Text}'.");
+
+            yield return chunk;
+        }
+    }
+
+    private static FrozenSet<string> CreatePredefinedSet(ReadOnlySpan<string> predefinedClasses, string fallbackClass)
+    {
+        if (predefinedClasses.Length == 0)
+        {
+            Throw.ArgumentException(nameof(predefinedClasses), "Predefined classes must be provided.");
+        }
+
+        HashSet<string> predefinedClassesSet = new(StringComparer.Ordinal) { fallbackClass };
+        foreach (string predefinedClass in predefinedClasses)
+        {
+#if NET
+            if (predefinedClass.Contains(',', StringComparison.Ordinal))
+#else
+            if (predefinedClass.IndexOf(',') >= 0)
+#endif
+            {
+                Throw.ArgumentException(nameof(predefinedClasses), $"Predefined class '{predefinedClass}' must not contain ',' character.");
+            }
+
+            if (!predefinedClassesSet.Add(predefinedClass))
+            {
+                if (predefinedClass.Equals(fallbackClass, StringComparison.Ordinal))
+                {
+                    Throw.ArgumentException(nameof(predefinedClasses), $"Fallback class '{fallbackClass}' must not be one of the predefined classes.");
+                }
+
+                Throw.ArgumentException(nameof(predefinedClasses), $"Duplicate class found: '{predefinedClass}'.");
+            }
+        }
+
+        return predefinedClassesSet.ToFrozenSet();
+    }
+
+    private static ChatOptions CreateChatOptions(ReadOnlySpan<string> predefinedClasses, ChatOptions? userProvided, string fallbackClass)
+    {
+        StringBuilder sb = new("You are a classification expert. Analyze the given text and assign a single, most relevant class. ");
+
+#pragma warning disable IDE0058 // Expression value is never used
+        sb.Append("Use only the following predefined classes: ");
+        for (int i = 0; i < predefinedClasses.Length; i++)
+        {
+            sb.Append(predefinedClasses[i]);
+            if (i < predefinedClasses.Length - 1)
+            {
+                sb.Append(", ");
+            }
+        }
+
+        sb.Append(" and return ").Append(fallbackClass).Append(" when unable to classify.");
+#pragma warning restore IDE0058 // Expression value is never used
+
+        ChatOptions result = userProvided?.Clone() ?? new();
+        result.Instructions = sb.ToString();
```

Comment:
Should we incorporate any instructions already set? Or you actually want to overwrite?